src/daemon.c

```diff
@@ -293,15 +291,22 @@ int become_daemon(int dont_fork, int close_all_files, const char *user, const ch
 		close(dev_null);
 
 	// generate our pid file
-	{
+	if(pidfile[0]) {
 		unlink(pidfile);
```

Comment:
oh! just saw this... you must not unlink pidfile, as you do not have permission to create a new file.